x-pack/plugins/transform/server/routes/api/transforms.ts

```diff
@@ -165,15 +160,13 @@ export function registerTransformsRoutes(routeDependencies: RouteDependencies) {
     },
     license.guardApiRoute<TransformIdParamSchema, undefined, undefined>(async (ctx, req, res) => {
       const { transformId } = req.params;
-      const options = {
-        ...(transformId !== undefined ? { transformId } : {}),
-      };
       try {
-        const stats = await ctx.transform!.dataClient.callAsCurrentUser(
-          'transform.getTransformsStats',
-          options
-        );
-        return res.ok({ body: stats });
+        const {
+          body,
+        } = await ctx.core.elasticsearch.client.asCurrentUser.transform.getTransformStats({
+          transform_id: transformId,
```

Comment:
the previous version wouldn't send a transform Id it it was undefined, this now always sends one.
is this change intentional?

Reply:
yes, this route has this validation schema https://github.com/elastic/kibana/blob/dd1822047c86239769f17d89799ace47c58e6ee6/x-pack/plugins/transform/common/api_schemas/common.ts#L35, hence this check is redundant

Reply:
also covered by API integration tests and checked manually